File: backend/routers/employees.py
from fastapi import HTTPException, APIRouter
from pydantic import BaseModel
from database import get_connection

router = APIRouter(prefix="/employees", tags=["employees"])

class EmployeeCreate(BaseModel):
    name: str
    email: str

class Employee(EmployeeCreate):
    id: int

@router.get("/", response_model=list[Employee])
def get_employees():
    query = "SELECT * FROM employees"
    with get_connection() as conn:
        with conn.cursor() as cur:
            cur.execute(query)
            employees = cur.fetchall()
            return employees


@router.post("/", response_model=Employee)
def create_employee(employee: EmployeeCreate):
    query = """
        INSERT INTO employees (name, email)
        VALUES (%s, %s)
        RETURNING id, name, email
        """
    with get_connection() as conn:
        with conn.cursor() as cur:
            cur.execute(query, (employee.name, employee.email))
            new_employee = cur.fetchone()
            if new_employee is None:
                raise HTTPException(status_code=400, detail="Failed to create employee")
            return new_employee

File: frontend/db-frontend/api/employees.ts
const API_BASE_URL = 'http://127.0.0.1:8000';

export type employeeCreate = {
    name: string;
    email: string;
}

export const fetchEmployees = async () => {
    try {
        const response = await fetch(`${API_BASE_URL}/employees/`);
        if (!response.ok) {
            throw new Error('Failed to fetch employees');
        }
        const employees = await response.json();
        return employees;
    } catch (error) {
        console.error('Error fetching employees:', error);
        throw error;
    }
}

export const createEmployee = async (data: employeeCreate) => {
    try {
        const response = await fetch(`${API_BASE_URL}/employees/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(data),
        });
        if (!response.ok) {
            throw new Error('Failed to create employee');
        }
        const newEmployee = await response.json();
        return newEmployee;
    } catch (error) {
        console.error('Error creating employee:', error);
        throw error;
    }
}